--- a/arch/powerpc/boot/types.h
+++ b/arch/powerpc/boot/types.h
@@ -40,6 +40,9 @@
 #define min_t(type, a, b) min(((type) a), ((type) b))
 #define max_t(type, a, b) max(((type) a), ((type) b))

+/* with C23, defining bool is now obsolete, and will cause an error. */
+#if __STDC_VERSION__ < 202311L
+
 typedef int bool;

 #ifndef true
@@ -49,4 +52,7 @@
 #ifndef false
 #define false 0
 #endif
+
+#endif
+
 #endif /* _TYPES_H_ */