--- ./types.h.orig	2026-01-04 19:17:16.972273000 +0000
+++ ./types.h	2026-01-04 19:19:07.060149000 +0000
@@ -40,6 +40,9 @@
 #define min_t(type, a, b) min(((type) a), ((type) b))
 #define max_t(type, a, b) max(((type) a), ((type) b))

+/* with C23, defining bool is now obsolete, and will cause an error. */
+#if __STDC_VERSION__ < 202311L
+
 typedef int bool;

 #ifndef true
@@ -49,4 +52,7 @@
 #ifndef false
 #define false 0
 #endif
+
+#endif
+
 #endif /* _TYPES_H_ */